### PR TITLE
Provide initial tree filter (matching commands)

### PR DESCRIPTION
**Pull request by layderv**
For this PR see: https://github.com/hishamhm/htop/issues/836#issuecomment-480027645

### CODE REVIEW

File: Settings.h [hunk 1]
```diff
@@ -3,12 +3,13 @@
 /*
 htop - Settings.h
 (C) 2004-2011 Hisham H. Muhammad
-Released under the GNU GPLv2, see the COPYING file
+Released under the GNU GPL, see the COPYING file
```
Comment (fasterit) [inline Settings.h:6]: rebase to HEAD please

**Review by fasterit — CHANGES_REQUESTED** [review]
I think this should integrate with the Filter functionality (F4 key), so that -F pre-sets a filter that can be cleared / amended in the UI. If that filter is regex, it should be regex in both use cases to not confuse users.
To me just plain string matching would be fine as well.

**Comment by layderv** [comment]
> I think this should integrate with the Filter functionality (F4 key), so that -F pre-sets a filter that can be cleared / amended in the UI. If that filter is regex, it should be regex in both use cases to not confuse users.
> To me just plain string matching would be fine as well.

@fasterit Your comment make sense, this PR implements it. (By "-F pre-sets..." I think you meant "-P" as in the option.)

**Comment by fasterit** [comment]
> @fasterit Your comment make sense, this PR implements it. (By "-F pre-sets..." I think you meant "-P" as in the option.)

No, I meant -F / --filter as that is consistent with the existing Filter (F4) GUI option.

**Comment by fasterit** [comment]
The second iteration looks good except for

- Old license header (work of a fresh master / HEAD please)
- Command line flag should be -F / --filter
- Man page needs update

**Comment by layderv** [comment]
@fasterit updated, thank you for the comments

**Comment by fasterit** [comment]
@layderv Excellent PR. Merged. Thank you very much.

**Comment by layderv** [comment]
@fasterit would it be possible to approve the PR also? :)

**Comment by fasterit** [comment]
I can't click approve on a PR that is already merged and thus closed.
Merging is the best approval you can get as it confirms the quality of your work.

**Comment by layderv** [comment]
@fasterit ok, I was asking because for the hacktoberfest the PR is pending with reason "Your PR has not yet been accepted by a maintainer. A maintainer can accept your PR by merging it, adding the "hacktoberfest-accepted" label to it, or approving it" (and I believe the update should happen within seconds). Not sure why, but no problem :)

**Comment by fasterit** [comment]
hm, "your PR by merging it". I've add the label, just in case.

**Comment by layderv** [comment]
It worked, thank you! @fasterit

**Pull request closed without merging**